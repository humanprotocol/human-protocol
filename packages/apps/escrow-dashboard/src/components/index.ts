export * from './App';
export * from './Footer';
export * from './Header';
export * from './Layout';

export * from './Cards';
export * from './ConnectButton';
export * from './CopyAddressButton';
export * from './Icons';
export * from './NetworkSelect';
export * from './PageWrapper';
export * from './SearchBox';
export * from './SocialIcons';
export * from './ViewTitle';
export * from './WalletModal';

export * from './Escrow';
export * from './Faucet';
export * from './Kvstore';
export * from './Leaderboard';
export * from './News';
export * from './SolvedTasks';
export * from './Token';
