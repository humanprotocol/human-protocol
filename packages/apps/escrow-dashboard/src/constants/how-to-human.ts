export const HOW_TO_HUMAN_DATA = [
  {
    key: 'introduction',
    label: 'Introduction',
    title: 'What is HUMAN Protocol?',
    youtubeId: '-x_kZSUc3b8',
  },
  {
    key: 'types-of-jobs',
    label: 'Types of Jobs',
    title: 'Job Types on HUMAN Protocol',
    youtubeId: 'wHtdQZkxPNc',
  },
];

export default HOW_TO_HUMAN_DATA;
