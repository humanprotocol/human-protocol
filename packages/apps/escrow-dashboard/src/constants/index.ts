import { ChainId } from '@human-protocol/sdk';

export const V2_SUPPORTED_CHAIN_IDS = [
  ChainId.MAINNET,
  ChainId.GOERLI,
  ChainId.BSC_MAINNET,
  ChainId.BSC_TESTNET,
  ChainId.POLYGON,
  ChainId.POLYGON_MUMBAI,
  ChainId.MOONBEAM,
  ChainId.MOONBASE_ALPHA,
  ChainId.CELO,
  ChainId.CELO_ALFAJORES,
  // ChainId.SKALE,
  // ChainId.AVALANCHE,
  // ChainId.AVALANCHE_TESTNET,
];

export const SUPPORTED_CHAIN_IDS = [
  ChainId.MAINNET,
  ChainId.RINKEBY,
  ChainId.GOERLI,
  ChainId.BSC_MAINNET,
  ChainId.BSC_TESTNET,
  ChainId.POLYGON,
  ChainId.POLYGON_MUMBAI,
  ChainId.SKALE,
  ChainId.MOONBEAM,
  ChainId.MOONBASE_ALPHA,
  ChainId.AVALANCHE_TESTNET,
  ChainId.AVALANCHE,
  ChainId.CELO,
  ChainId.CELO_ALFAJORES,
];

export const L1_L2_CHAIN_IDS = [
  ChainId.BSC_MAINNET,
  ChainId.POLYGON,
  ChainId.SKALE,
  ChainId.MOONBEAM,
  ChainId.AVALANCHE,
];

export const TESTNET_CHAIN_IDS = [
  ChainId.RINKEBY,
  ChainId.GOERLI,
  ChainId.BSC_TESTNET,
  ChainId.POLYGON_MUMBAI,
  ChainId.MOONBASE_ALPHA,
  ChainId.AVALANCHE_TESTNET,
  ChainId.CELO_ALFAJORES,
];

export const FAUCET_CHAIN_IDS = [
  ChainId.GOERLI,
  ChainId.BSC_TESTNET,
  ChainId.POLYGON_MUMBAI,
  ChainId.MOONBASE_ALPHA,
  ChainId.AVALANCHE_TESTNET,
  ChainId.SKALE,
  ChainId.CELO_ALFAJORES,
];

export const RPC_URLS: {
  [chainId in ChainId]?: string;
} = {
  [ChainId.MAINNET]:
    'https://eth-mainnet.g.alchemy.com/v2/VVDrD3TpJv8ZBP4CiwH2m5Oj6r0hM2st',
  [ChainId.GOERLI]:
    'https://goerli.infura.io/v3/9aa3d95b3bc440fa88ea12eaa4456161',
  [ChainId.BSC_MAINNET]: 'https://bsc-dataseed1.binance.org/',
  [ChainId.BSC_TESTNET]: 'https://bsc-testnet.publicnode.com',
  [ChainId.POLYGON]: 'https://polygon-rpc.com/',
  [ChainId.POLYGON_MUMBAI]: 'https://rpc-mumbai.maticvigil.com',
  [ChainId.MOONBEAM]: 'https://rpc.api.moonbeam.network',
  [ChainId.MOONBASE_ALPHA]: 'https://rpc.api.moonbase.moonbeam.network',
  [ChainId.AVALANCHE_TESTNET]: 'https://api.avax-test.network/ext/C/rpc',
  [ChainId.AVALANCHE]: 'https://api.avax.network/ext/bc/C/rpc',
  [ChainId.SKALE]: 'https://mainnet.skalenodes.com/v1/wan-red-ain',
  [ChainId.CELO_ALFAJORES]: 'https://alfajores-forno.celo-testnet.org',
  [ChainId.CELO]: 'https://forno.celo.org',
};

export const FAST_INTERVAL = 10_000;
export const SLOW_INTERVAL = 60_000;

export const ROLES = [
  'Validator',
  'Operator (Job Launcher)',
  'Exchange Oracle',
  'Reputation Oracle',
  'Recording Oracle',
];

export const HM_TOKEN_DECIMALS = 18;

export const STAKING_CONTRACT_ADDRESS =
  '0x1fA701df2bb75f2cE8B6439669BD1eCfCf8b26fe';

export const BITFINEX_SUPPORTED_CHAIN_IDS = [ChainId.MAINNET, ChainId.POLYGON];

export const BITFINEX_HOT_WALLET_ADDRESS =
  '0x77134cbc06cb00b66f4c7e623d5fdbf6777635ec';
