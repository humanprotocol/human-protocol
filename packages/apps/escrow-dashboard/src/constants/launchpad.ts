import imageBinaryImg from '../assets/launchpad/task-types/image_binary.png';
import imageBoundingBoxImg from '../assets/launchpad/task-types/image_bounding_box.png';
import imageMultipleChoiceImg from '../assets/launchpad/task-types/image_multiple_choice.png';
import imageRedoBBImg from '../assets/launchpad/task-types/image_redo_bb.png';
import imageSemanticSegmentationImg from '../assets/launchpad/task-types/image_semantic_segmentation.png';
import marketMakingHuFiImg from '../assets/launchpad/task-types/market_making_hufi.png';
import openQueriesMarketResearchImg from '../assets/launchpad/task-types/open_queries_market_research.png';
import textFreeEntryOcrImg from '../assets/launchpad/task-types/text_free_entry_ocr.png';
import textMultipleChoiceImg from '../assets/launchpad/task-types/text_multiple_choice.png';

export enum TextLabelingJobType {
  FREE_ENTRY_OCR = 'FREE_ENTRY_OCR',
  MULTIPLE_CHOICE = 'TEXT_MULTIPLE_CHOICE',
}

export enum ImageLabelingJobType {
  BINARY = 'BINARY',
  BOUNDING_BOX = 'BOUNDING_BOX',
  REDO_BB = 'REDO_BB',
  MULTIPLE_CHOICE = 'IMAGE_MULTIPLE_CHOICE',
  SEMANTIC_SEGMENTATION = 'SEMANTIC_SEGMENTATION',
}

export enum MarketMakingJobType {
  HUFI = 'HUFI',
}

export enum OpenQueriesJobType {
  MARKET_RESEARCH = 'MARKET_RESEARCH',
}

export const TextLabelingJobTypeData = {
  [TextLabelingJobType.FREE_ENTRY_OCR]: {
    label: 'Text free entry/OCR',
    description:
      'Type the words shown (with image) or just text. Name an object that tells time.',
    image: textFreeEntryOcrImg,
  },
  [TextLabelingJobType.MULTIPLE_CHOICE]: {
    label: 'Multiple Choice',
    description: 'Click the closest match to the words shown.',
    image: textMultipleChoiceImg,
  },
};

export const ImageLabelingJobTypeData = {
  [ImageLabelingJobType.BINARY]: {
    label: 'Binary',
    description: 'Click all the cats.',
    image: imageBinaryImg,
  },
  [ImageLabelingJobType.BOUNDING_BOX]: {
    label: 'Bounding box',
    description: 'Click the face or draw a box on the face.',
    image: imageBoundingBoxImg,
  },
  [ImageLabelingJobType.REDO_BB]: {
    label: 'Redo BB',
    description: 'Reposition this box to better enclose the face.',
    image: imageRedoBBImg,
  },
  [ImageLabelingJobType.MULTIPLE_CHOICE]: {
    label: 'Multiple Choice',
    description: 'Click all the words that apply to this image.',
    image: imageMultipleChoiceImg,
  },
  [ImageLabelingJobType.SEMANTIC_SEGMENTATION]: {
    label: 'Semantic segmentation',
    description:
      'Paint the areas of the image that correspond to this subject.',
    image: imageSemanticSegmentationImg,
  },
};

export const MarketMakingJobTypeData = {
  [MarketMakingJobType.HUFI]: {
    label: 'HuFi',
    description:
      'Launch campaigns, choose trading pairs, and select exchange platforms.',
    image: marketMakingHuFiImg,
  },
};

export const OpenQueriesJobTypeData = {
  [OpenQueriesJobType.MARKET_RESEARCH]: {
    label: 'Market research',
    description:
      'Open queries with an open text answer box. Query can contain a link.',
    image: openQueriesMarketResearchImg,
  },
};
