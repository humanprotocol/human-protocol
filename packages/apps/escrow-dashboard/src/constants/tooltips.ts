export const TOOLTIPS = {
  TASKS: 'Average solved tasks by workers',
  WORKERS: 'Number of daily active workers',
  PAYMENTS: 'All bulk paid out HMT amount',
  TRANSACTIONS: 'Daily on-chain HMT transaction count',
  SOLVED_TASKS: 'Number of tasks solved by workers',
  TOTAL_TRANSFERS: 'Total HMT transfer count',
  HOLDERS:
    'Number of HMT holders, this number is all time number and not for the time period selected',
  TOTAL_SUPPLY: 'Total supply of HMT',
};
