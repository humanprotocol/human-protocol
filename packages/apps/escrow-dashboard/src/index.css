body {
    margin: 0;
    font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Inter', 'Oxygen',
    'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue',
    sans-serif;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
}

code {
    font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New',
    monospace;
}


.pubkey::-webkit-scrollbar {
    width: 12px;
    background-color: #F5F5F5;
}

.pubkey::-webkit-scrollbar-thumb {
    border-radius: 10px;
    -webkit-box-shadow: inset 0 0 6px rgba(0, 0, 0, .3);
    background-color: #320A8D;
}