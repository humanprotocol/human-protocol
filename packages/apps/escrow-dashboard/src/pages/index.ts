export * from './EscrowDetail';
export * from './Faucet';
export * from './Launchpad';
export * from './LaunchpadExploreJobs';
export * from './LeaderDetail';
export * from './Kvstore';
export * from './Main';
export * from './Owner';
export * from './Profile';
export * from './MyHMT';
export * from './ConfigureOracle';
export * from './HowToHuman';
