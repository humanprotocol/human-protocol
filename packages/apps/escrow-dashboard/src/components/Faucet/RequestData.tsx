import { ChainId, NETWORKS, NetworkData } from '@human-protocol/sdk';
import {
  Button,
  FormControl,
  Grid,
  InputLabel,
  MenuItem,
  Paper,
  Select,
  Snackbar,
  TextField,
  Typography,
} from '@mui/material';
import { Dispatch, FC, useState } from 'react';
import { Link } from 'react-router-dom';
import { Alert } from '../Alert';
import { FAUCET_CHAIN_IDS } from 'src/constants';

export type RequestDataProps = {
  step: number;
  setStep: Dispatch<number>;
  setTxHash: Dispatch<string>;
  network: NetworkData;
  setNetwork: Dispatch<NetworkData>;
};

export const RequestData: FC<RequestDataProps> = ({
  step,
  setStep,
  setTxHash,
  network,
  setNetwork,
}) => {
  const [address, setAddress] = useState<string>('');
  const [error, setError] = useState<string>('');

  const handleSubmit = async () => {
    setStep(1);
    const payload = { address: address, chainId: network.chainId };
    const response = await fetch(
      `${import.meta.env.VITE_APP_FAUCET_SERVER_URL}/faucet`,
      {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
          Accept: 'application/json',
        },
        body: JSON.stringify(payload),
      }
    );

    const result = await response.json();
    if (!result.status) {
      setError(result.message);
      setStep(0);
    } else {
      setTxHash(result.txHash);
      setStep(2);
    }
  };

  return (
    <Paper>
      <Snackbar
        anchorOrigin={{ vertical: 'top', horizontal: 'center' }}
        open={error.length > 0}
        autoHideDuration={3000}
        onClose={() => setError('')}
      >
        <Alert
          onClose={() => setError('')}
          severity="error"
          sx={{ width: '100%' }}
        >
          {error}
        </Alert>
      </Snackbar>
      <Grid
        container
        justifyContent="center"
        direction="column"
        sx={{
          padding: { xs: 2, sm: 2, md: 10, lg: 10 },
          minWidth: 600,
        }}
      >
        <Grid item container direction="row" alignItems="center">
          <FormControl fullWidth disabled={step === 1}>
            <InputLabel>Network</InputLabel>
            <Select
              label="Network"
              variant="outlined"
              value={network.chainId}
              onChange={(e) =>
                setNetwork(NETWORKS[Number(e.target.value) as ChainId]!)
              }
            >
              {FAUCET_CHAIN_IDS.map((chainId) => (
                <MenuItem key={chainId} value={chainId}>
                  {NETWORKS[chainId]?.title}
                </MenuItem>
              ))}
            </Select>
          </FormControl>
        </Grid>

        <Grid
          item
          container
          direction="row"
          alignItems="center"
          sx={{
            marginTop: { xs: 1, sm: 1, md: 5, lg: 5 },
          }}
        >
          <TextField
            fullWidth
            id="outlined-basic"
            label="Address"
            variant="outlined"
            value={address}
            onChange={(e) => setAddress(e.target.value)}
            disabled={step === 1}
          />
        </Grid>
        <Grid
          item
          container
          direction="row"
          justifyContent="center"
          alignItems="flex-end"
          sx={{
            marginTop: { xs: 1, sm: 1, md: 5, lg: 5 },
          }}
        >
          <Button
            fullWidth
            onClick={handleSubmit}
            variant="contained"
            disabled={step === 1}
            sx={{
              '&.Mui-disabled': {
                background: '#2F0087',
                color: '#FFFFFF',
              },
            }}
          >
            {step === 1
              ? network?.chainId === ChainId.SKALE
                ? 'Mining sFUEL'
                : 'Sending tokens'
              : network?.chainId === ChainId.SKALE
              ? 'Send sFUEL'
              : 'Send me'}
          </Button>
        </Grid>
        <Grid
          item
          container
          direction="row"
          justifyContent="center"
          alignItems="flex-end"
          sx={{
            marginTop: { xs: 1, sm: 1, md: 5, lg: 5 },
          }}
        >
          <Typography
            variant="body2"
            sx={
              step === 1
                ? {
                    fontWeight: 'bold',
                    color: (theme) => theme.palette.text.disabled,
                  }
                : {
                    fontWeight: 'bold',
                  }
            }
          >
            {network?.chainId === ChainId.SKALE
              ? 'Will top up sFUEL balance on SKALE'
              : 'Token address:'}
          </Typography>
        </Grid>
        <Grid
          item
          container
          direction="row"
          justifyContent="center"
          alignItems="flex-end"
        >
          <Typography
            variant="body2"
            sx={
              step === 1
                ? {
                    mt: 1,
                    color: (theme) => theme.palette.text.disabled,
                  }
                : {
                    mt: 1,
                  }
            }
          >
            <Link
              to={
                network?.chainId === ChainId.SKALE
                  ? network?.scanUrl
                  : network?.scanUrl + '/address/' + network?.hmtAddress
              }
              style={{ textDecoration: 'none', color: 'inherit' }}
            >
              {network?.chainId === ChainId.SKALE
                ? 'Link to SKALE Explorer'
                : network?.hmtAddress}
            </Link>
          </Typography>
        </Grid>
      </Grid>
    </Paper>
  );
};
