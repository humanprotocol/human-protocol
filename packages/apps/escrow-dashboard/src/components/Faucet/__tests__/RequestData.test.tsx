import { render, screen } from '@testing-library/react';
import { act } from 'react-dom/test-utils';
import { MemoryRouter } from 'react-router-dom';
import { create } from 'react-test-renderer';

import { RequestData, RequestDataProps } from '../RequestData';

import { ChainId, ESCROW_NETWORKS } from 'src/constants';

describe('when rendered AfterConnect component', () => {
  it('should render `text` prop', async () => {
    await act(async () => {
      render(
        <RequestData
          {...({} as RequestDataProps)}
          network={ESCROW_NETWORKS[ChainId.POLYGON_MUMBAI]!}
        />,
        { wrapper: MemoryRouter }
      );
    });
    expect(screen.getByText(/Token address:/)).toBeInTheDocument();
  });
});

it('AfterConnect component renders correctly, corresponds to the snapshot', () => {
  const tree = create(
    <MemoryRouter>
      <RequestData
        {...({} as RequestDataProps)}
        network={ESCROW_NETWORKS[ChainId.POLYGON_MUMBAI]!}
      />
    </MemoryRouter>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
