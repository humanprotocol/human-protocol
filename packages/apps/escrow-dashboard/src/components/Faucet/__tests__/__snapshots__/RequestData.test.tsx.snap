// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AfterConnect component renders correctly, corresponds to the snapshot 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1ps6pg7-MuiPaper-root"
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-s2aonh-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-item css-lx31tv-MuiGrid-root"
    >
      <div
        className="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink={true}
        >
          Network
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
          onClick={[Function]}
        >
          <div
            aria-controls=":r0:"
            aria-expanded="false"
            aria-haspopup="listbox"
            className="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            role="combobox"
            tabIndex={0}
          >
            Polygon Mumbai
          </div>
          <input
            aria-hidden={true}
            aria-invalid={false}
            className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            disabled={false}
            onAnimationStart={[Function]}
            onChange={[Function]}
            required={false}
            tabIndex={-1}
            value={80001}
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-14lo706"
            >
              <span>
                Network
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-item css-12db1rc-MuiGrid-root"
    >
      <div
        className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink={false}
          htmlFor="outlined-basic"
          id="outlined-basic-label"
        >
          Address
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            disabled={false}
            id="outlined-basic"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="text"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-yjsfm1"
            >
              <span>
                Address
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-item css-fft5ex-MuiGrid-root"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth css-1po1fcj-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Send me
      </button>
    </div>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-item css-fft5ex-MuiGrid-root"
    >
      <p
        className="MuiTypography-root MuiTypography-body2 css-1d3njat-MuiTypography-root"
      >
        Token address:
      </p>
    </div>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-item css-1w0t266-MuiGrid-root"
    >
      <p
        className="MuiTypography-root MuiTypography-body2 css-1xwhmis-MuiTypography-root"
      >
        <a
          href="https://mumbai.polygonscan.com/address/0x0376D26246Eb35FF4F9924cF13E6C05fd0bD7Fb4"
          style={
            {
              "color": "inherit",
              "textDecoration": "none",
            }
          }
        >
          0x0376D26246Eb35FF4F9924cF13E6C05fd0bD7Fb4
        </a>
      </p>
    </div>
  </div>
</div>
`;
